database/src/main/postgres/runs/V1.8.3__get_partitioning_checkpoints.sql

```diff
@@ -86,28 +102,28 @@ BEGIN
             md.measured_columns,
             M.measurement_value,
             C.process_start_time AS checkpoint_start_time,
-            C.process_end_time AS checkpoint_end_time
+            C.process_end_time AS checkpoint_end_time,
+            (SELECT COUNT(*) > i_limit FROM limited_checkpoints) AS has_more
         FROM
             runs.checkpoints C
-        JOIN
+                JOIN
             runs.measurements M ON C.id_checkpoint = M.fk_checkpoint
-        JOIN
+                JOIN
             runs.measure_definitions MD ON M.fk_measure_definition = MD.id_measure_definition
         WHERE
-            C.fk_partitioning = _fk_partitioning
-        AND
-            (i_checkpoint_name IS NULL OR C.checkpoint_name = i_checkpoint_name)
+            C.id_checkpoint IN (SELECT LC.id_checkpoint FROM limited_checkpoints as LC WHERE LC.rn <= i_limit)
         ORDER BY
             C.process_start_time,
-            C.id_checkpoint
-        LIMIT nullif(i_limit, 0);
+            C.id_checkpoint;
```

Comment:
**Maybe premature optimization:**
Would consider having two branches of `RETURN QUERY` based if `i_limit` is NULL or not. (Without `i_limit` `WITH` is not required.